webapi/Plugins/Chat/ApiConnectorPlugin.cs

```diff
@@ -0,0 +1,166 @@
+    //     class.
+    //
+    // Parameters:
+    //   bearerToken:
+    //     The bearer token to received by the WebAPI and used to obtain a new access token using the OBO Flow.
+    //
+    //   clientFactory:
+    //     The factory to use to create HttpClient instances.
+    //
+    //   PlannerOptions.OboOptions:
+    //     Configuration for the plugin defined in appsettings.json.
+    public ApiConnectorPlugin(string bearerToken, IHttpClientFactory clientFactory, PlannerOptions.OboOptions? onBehalfOfAuth, ILogger logger)
+    {
+        this._bearerToken = bearerToken ?? throw new ArgumentNullException(bearerToken);
+        this._clientFactory = clientFactory;
+        this._logger = logger;
+
+        this._clientId = onBehalfOfAuth?.ClientId ?? throw new ArgumentNullException(onBehalfOfAuth?.ClientId);
+        this._clientSecret = onBehalfOfAuth?.ClientSecret ?? throw new ArgumentNullException(onBehalfOfAuth?.ClientSecret);
+        this._tenantId = onBehalfOfAuth?.TenantId ?? throw new ArgumentNullException(onBehalfOfAuth?.TenantId);
+        this._authority = onBehalfOfAuth?.Authority ?? throw new ArgumentNullException(onBehalfOfAuth?.Authority);
+    }
+
+    //
+    // Summary:
+    //     Call a Graph API with the OData query and the Graph API Scopes based on user input.
+    //
+    // Parameters:
+    //   apiURL:
+    //     The URL of the GRAPH API with the OData query to call.
+    //
+    //   graphScopes:
+    //     The comma separated value string with the Graph API Scopes needed to execute the
+    //     call.
+    //
+    //   cancellationToken:
+    //     The cancellation token.
+    //
+    // Returns:
+    //     The response from the GRAPH API.
+    //
+    // Exceptions:
+    //   T:System.ArgumentNullException:
+    //     apiURL is null or empty.
+    //
+    //   T:System.ArgumentNullException:
+    //     graphScopes is null or empty.
+    //
+    //   T:System.Net.Http.HttpRequestException:
+    //     Failed to get token: {response.StatusCode}.
+    //
+    //   T:System.Net.Http.HttpRequestException:
+    //     Failed to get access token.
+    //
+    //   T:System.Net.Http.HttpRequestException:
+    //     Failed to get graph data: {graphResponse.StatusCode}.
+    [SKFunction]
+    [Description("Call a Graph API with the OData query and the Graph API Scopes based on user input")]
+    public async Task<string> CallGraphApiTasksAsync([Description("Url of the Graph API with the OData query to call")] string apiToCall, [Description("Comma separated value string with the Graph API Scopes needed to execute the call")] string graphScopes, CancellationToken cancellationToken = default)
+    {
+        if (string.IsNullOrEmpty(apiToCall))
+        {
+            throw new ArgumentNullException(apiToCall);
+        }
+
+        if (string.IsNullOrEmpty(graphScopes))
+        {
+            throw new ArgumentNullException(graphScopes);
+        }
+
+        string? accessToken = string.Empty;
```

Comment:
```suggestion
        string? accessToken = GetOboAccessToken(graphScopes);
```

Can you separate lines 126 - 143 into its own method called `GetOboAccessToken`? The isolated token acquisition will be easier to follow if someone is referencing this as a sample.